### PR TITLE
[cert-manager] Add `kube-lego` alternative

### PR DESCRIPTION
**Pull request by alebabai**
## what
* Added helmfile for Cert Manager chart

## why
* `kube-lego` deprecated

## credits
* Donated by @rohitverma of Niki.ai

## references
#1

### CODE REVIEW

**Comment by osterman** [comment]
```
[SYSTEM] Error: Failed to push image r.cfcr.io/cloudposse/helmfiles:0.1.1-feature/stable-grafana.sh
a.5773698; caused by Error: Failed to tag image r.cfcr.io/cloudposse/helmfiles@sha256:a704d21359983
c84e5b8f20bba1365635931839427b42352708b56ad6109138b to {"repository":"cloudposse/helmfiles","tag":"
0.1.1-feature/stable-grafana.sha.5773698","registry":"r.cfcr.io","imageName":"r.cfcr.io/cloudposse/
helmfiles:0.1.1-feature/stable-grafana.sha.5773698"}; caused by Error: (HTTP code 500) server error
 - invalid tag format    
```
I thought we fixed this in the `build-harness`? Maybe we are pinned to an old release in `helmfiles`.

If not, can you submit a fix in the `build-harness`?

File: helmfile.d/0120.cert-manager.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+helmDefaults:
+  args:
+    - "--wait"
+    - "--recreate-pods"
```
Comment (osterman) [inline helmfile.d/0120.cert-manager.yaml]: Lets drop this unless needed. Causes outages.

File: helmfile.d/0120.cert-manager.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    chart: "cert-manager"
+    component: "ingress"
+    namespace: "kube-system"
+    vendor: "jetstack"
+    default: "true"
+  chart: "stable/cert-manager"
+  version: "v0.4.0"
+  values:
+    - rbac:
+        create: true
```
Comment (osterman) [inline helmfile.d/0120.cert-manager.yaml]: Make this a parameter

**Review by osterman — CHANGES_REQUESTED** [review]
See comments

File: helmfile.d/0120.cert-manager.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+## CERT-MANAGER - Automatic Let's Encrypt for Ingress  #########################
+################################################################################
+
+- name: "cert-manager"
+  namespace: "kube-system"
+  labels:
+    chart: "cert-manager"
+    component: "ingress"
+    namespace: "kube-system"
+    vendor: "jetstack"
```
Comment (aknysh) [inline helmfile.d/0120.cert-manager.yaml:26]: vendor: "kubernetes"

File: helmfile.d/0120.cert-manager.yaml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    default: "true"
```
Comment (aknysh) [inline helmfile.d/0120.cert-manager.yaml]: default: "false"
we have two cert managers now (kube-lego id another one).
we need to make both of them not default

File: helmfile.d/0120.cert-manager.yaml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  chart: "stable/cert-manager"
+  version: "v0.4.0"
+  values:
+    - rbac:
+        create: true
+      ingressShim:
+        defaultIssuerName: "letsencrypt-prod"
```
Comment (aknysh) [inline helmfile.d/0120.cert-manager.yaml]: make these parameters

**Comment by aknysh** [comment]
add this section:

```
serviceAccount:
  # Specifies whether a service account should be created
  create: true
  # The name of the service account to use.
  # If not set and create is true, a name is generated using the fullname template
  name:
```

Make `create` and `name` parameters

**Comment by aknysh** [comment]
add `resources` section
similar to this:

```
resources:
        limits:
          cpu: "200m"
          memory: "256Mi"
        requests:
          cpu: "50m"
          memory: "128Mi"
```

**Review by aknysh — DISMISSED** [review]
inline comments

**Comment by osterman** [comment]
This will fix CI: https://github.com/cloudposse/helmfiles/pull/7

File: helmfile.d/0120.cert-manager.yaml (hunk 3 above)
Comment (osterman) [reply helmfile.d/0120.cert-manager.yaml:26]: No, the vendor is `jetstack`

Comment (osterman) [reply helmfile.d/0120.cert-manager.yaml:26]: https://github.com/jetstack/cert-manager

File: helmfile.d/0120.cert-manager.yaml (hunk 4 above)
Comment (osterman) [reply helmfile.d/0120.cert-manager.yaml]: No, `default` should indicate the one we prefer. The same is true for `kube2iam` and `kiam`. The new default is `kiam`.

**Review by osterman — APPROVED** [review]

File: helmfile.d/0120.cert-manager.yaml (hunk 4 above)
Comment (osterman) [reply helmfile.d/0120.cert-manager.yaml]: Ok, what I should have said, is we need one to be true. Let's make this one false and leave `kube-lego` the default until we document how to use `cert-manager.

**Review by osterman — DISMISSED** [review]
Fix this: https://github.com/cloudposse/helmfiles/pull/5/files#r207630689

**Review by osterman — APPROVED** [review]

**Pull request merged**